app/src/main/java/com/example/habitsmasher/listeners/SuccessListenerWithToast.java

```diff
@@ -0,0 +1,35 @@
+package com.example.habitsmasher.listeners;
+
+import android.content.Context;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnSuccessListener;
+/**
+ * This is the SuccessListenerWithToast class
+ * It implements the OnSuccessListener
+ * Its purpose is to perform the onSuccess listener actions
+ */
+public class SuccessListenerWithToast implements OnSuccessListener {
```

Comment:
If you choose to combine the FailureListener into one class with the boolean flag passed in as a parameter, I'd do the same to this one. Just another design decision to consider.